build/download_docker_binary.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+if [[ $# -ne 3 ]]; then
+    echo "Illegal number of parameters" >&2
+    exit 1
+fi
+
+PLATFORM=$1
+ARCH=$2
+DOCKER_VERSION=${3:1}
+DOWNLOAD_FOLDER=".tmp/download"
+
```

Comment:
Do you need an override for mac platforms?  as the path for mac is meant to be:
`https://download.docker.com/mac/ ...`

And not this:
`https://download.docker.com/darwin`

Reply:
won't need mac, just make it as same as agent